gstd/src/common/handlers.rs

```diff
@@ -25,50 +25,208 @@
+///     [`write!`] macro.
+///   - In stable Rust, we need to modify the default panic handler message
+///     format.
+/// - `panic-location`: it displays `'{message}', {location}`
+///
+/// Default panic handler message format (according to <https://github.com/rust-lang/rust/pull/112849>):
+/// - Rust  <1.73: `panicked at '{message}', {location}`
+/// - Rust >=1.73: `panicked at {location}:\n{message}`
+///
+/// We parse the output of `impl Display for PanicInfo<'_>` and
+/// then convert it to custom format: `'{message}', {location}`.
+///
+/// Here is a test to verify that the default panic handler message format
+/// has not changed:
+/// ```
+/// use std::panic::{self, PanicInfo};
+///
+/// const MESSAGE: &str = "message";
+///
+/// fn expected_format(panic_info: &PanicInfo) -> String {
+///     let location = panic_info.location().unwrap();
+///     format!("panicked at {location}:\n{MESSAGE}")
+/// }
+///
+/// panic::set_hook(Box::new(|panic_info| {
+///     assert_eq!(panic_info.to_string(), expected_format(panic_info));
+/// }));
+///
+/// let result = panic::catch_unwind(|| {
+///     panic!("{MESSAGE}");
+/// });
+/// assert!(result.is_err());
+/// ```
+pub mod panic_handler {
+    #[cfg(target_arch = "wasm32")]
+    mod internal {
+        use crate::ext;
+        use core::panic::PanicInfo;
 
-    #[cfg(not(feature = "debug"))]
-    #[panic_handler]
-    pub fn panic(_: &PanicInfo) -> ! {
-        ext::panic("no info")
-    }
+        /// Minimal panic handler.
+        #[cfg(not(any(feature = "panic-message", feature = "panic-location")))]
+        #[panic_handler]
+        pub fn panic(_: &PanicInfo) -> ! {
+            ext::panic("no info")
+        }
+
+        #[cfg(any(feature = "panic-message", feature = "panic-location"))]
+        mod constants {
+            /// Max amount of bytes allowed to be thrown as string explanation
+            /// of the error.
+            pub const TRIMMED_MAX_LEN: usize = 1024; //TODO: do not duplicate `gear_core::str::TRIMMED_MAX_LEN`
+            /// This prefix is used to print debug message:
+            /// `debug!("panic occurred: {msg}")`.
+            pub const PANIC_OCCURRED: &str = "panic occurred: ";
+            /// Size of array string that will be allocated on the stack.
+            pub const ARRAY_STRING_MAX_LEN: usize = if cfg!(feature = "debug") {
+                PANIC_OCCURRED.len()
+            } else {
+                0
+            } + TRIMMED_MAX_LEN;
+            /// This prefix is used by `impl Display for PanicInfo<'_>`.
+            #[cfg(not(feature = "nightly"))]
+            pub const PANICKED_AT: &str = "panicked at ";
+        }
+
+        #[cfg(any(feature = "panic-message", feature = "panic-location"))]
+        use constants::*;
+
+        /// Panic handler for nightly Rust.
+        #[cfg(feature = "nightly")]
+        #[cfg(any(feature = "panic-message", feature = "panic-location"))]
```

Comment:
It's not always whole nightly implied
```suggestion
        #[cfg(feature = "panic-info-message")]
        #[cfg(feature = "panic-handler")]
        #[cfg(any(feature = "panic-message", feature = "panic-location"))]
```

Reply:
same below

Reply:
63a341bba9b0b32f6fdf76d767af17a22003f606